header/core/listener.go

```diff
@@ -20,24 +21,30 @@ import (
 // broadcasts the new `ExtendedHeader` to the header-sub gossipsub
 // network.
 type Listener struct {
-	bcast     header.Broadcaster
-	fetcher   *core.BlockFetcher
-	bServ     blockservice.BlockService
+	fetcher *core.BlockFetcher
+	bServ   blockservice.BlockService
+
 	construct header.ConstructFn
-	cancel    context.CancelFunc
+
+	broadcastExtendedHeader header.Broadcaster
```

Comment:
last optional nit (and I know i suggested this naming) but i realised it's bit ugly to do `broadcastExtendedHeader.Broadcast` so let's do something like `extHeaderBroadcaster.Broadcast`